assembl/static2/js/app/components/common/richTextEditor/attachmentsPlugin.jsx

```diff
@@ -0,0 +1,161 @@
+// @flow
+/* draft-js plugin for attachment management */
+import { convertFromRaw, convertToRaw, Entity, Modifier, RawContentState, SelectionState } from 'draft-js';
+import type { ContentBlock } from 'draft-js';
+import type { Attachment, Document } from '../attachments';
+
+const ENTITY_TYPE = 'document';
+const BLOCK_TYPE = 'atomic';
+
+type NodeType = Object;
+
+const plugin = {
```

Comment:
I can't use the existing plugins because we have a lot of use cases that they don't handled. For example, I need to be able to add non images attachments. It would have been a lot easier/quicker to use the existing plugin (and drag-n-drop and other stuff would have been simple to add).

No docs really here, it is not really a "plugin", I just use the patterns that I saw in draft-js-plugins